### PR TITLE
Signup: Cleanup main component

### PR DESCRIPTION
**Pull request by jsnmoon**
For the most part, this only moves around existing functions for readability and clarity. This is in preparation for my task for [reduxifying the signup flow](https://github.com/Automattic/wp-calypso/issues/6709).

A few instances where behavior has been tweaked:
1. Analytic events were moved from `componentWillMount` to `componentDidMount`. Note that componentWillMount is being [deprecated](https://reactjs.org/docs/react-component.html#unsafe_componentwillmount).
2. Redundant render helper functions were moved inside the main `render()` block.

To summarize, no new behavior has been intended. Any behavioral changes introduced with this PR should be considered a bug.

# Test Instructions
1. Audit the code.
2. Spin up this branch locally.
3. While logged out, go through user registration flow from `/`.
4. (For e2e testers) Try going through the registration flow for Jetpack and WooCommerce.

### CODE REVIEW

**Comment by alisterscott** [comment]
I did some sign up testing of this in IE11 and Edge - coudn't see any issues

@bsessions85 / @brbrr / @abidhahmed would you mind running some Jetpack / WooCommerce connect/sign up tests against this branch/PR?

File: client/signup/main.jsx [hunk 1]
```diff
@@ -77,6 +77,15 @@ class Signup extends React.Component {
...
+		isLoggedIn: PropTypes.bool,
+		loadTrackingTool: PropTypes.func.isRequired,
+		setSurvey: PropTypes.func.isRequired,
+		signupDependencies: PropTypes.object,
+		trackAffiliateReferral: PropTypes.func.isRequired,
+	};
+
 	constructor( props, context ) {
 		super( props, context );
 		SignupDependencyStore.setReduxStore( context.store );
```
Comment (jsnajdr) [inline client/signup/main.jsx]: An idea for another cleanup: for Flux/Redux interaction, we use `lib/redux-bridge` and its `reduxGetState` and `reduxDispatch` exports. Disclaimer: I'm not sure it really makes sense for signup, which has standalone Redux store different from the rest of Calypso.

**Review by jsnajdr — APPROVED** [review]
The code changes look good. Thanks for creating individual easy to understand commits.

The e2e tests are failing though, because of a Node version mismatch. Rebasing this branch onto the latest master and rerunning should help.

Approved if the e2e tests are 💚

Some ideas for further cleanups: get rid of `componentWillMount` and `componentWillReceiveProps`. They will become deprecated in React soon, as they are not compatible with async mode. Can their bodies be moved to `componentDidMount` and `componentDidUpdate`?

File: client/signup/main.jsx (hunk 1 above)
Comment (jsnmoon) [reply client/signup/main.jsx]: Good to know - I'd been wondering what `redux-bridge` was. 

Sign up has a lot of tech debt besides being written in Flux, so just slapping on `redux-bridge` might not be sufficient. 

I'd gone ahead and given this part of the codebase some tender loving care the past few days, so keep a lookout for future refactor/reduxification PRs :)

**Comment by jsnmoon** [comment]
> Some ideas for further cleanups: get rid of `componentWillMount` and `componentWillReceiveProps`. They will become deprecated in React soon, as they are not compatible with async mode. Can their bodies be moved to `componentDidMount` and `componentDidUpdate`? (@jsnajdr)

I absolutely agree with this direction. I'm not sure if they can be plucked into their `*Did*` counterparts, but it's worth exploring in a future PR.
